File: WikiCodeParser/Nodes/HtmlNode.cs
using System;
using System.Collections.Generic;

namespace WikiCodeParser.Nodes
{
    /// <summary>
    /// Html content. Has HTML before and after, and a node containing the content.
    /// </summary>
    public class HtmlNode : INode
    {
        public string HtmlBefore { get; }
        public INode Content { get; set; }
        public string HtmlAfter { get; }

        public string PlainBefore { get; set; }
        public string PlainAfter { get; set; }

        public HtmlNode(string htmlBefore, INode content, string htmlAfter)
        {
            HtmlBefore = htmlBefore;
            Content = content;
            HtmlAfter = htmlAfter;
            PlainBefore = PlainAfter = String.Empty;
        }

        public string ToHtml()
        {
            return HtmlBefore + Content.ToHtml() + HtmlAfter;
        }

        public string ToPlainText()
        {
            return PlainBefore + Content.ToPlainText() + PlainAfter;
        }

        public IList<INode> GetChildren()
        {
            return new[] {Content};
        }

        public void ReplaceChild(int i, INode node)
        {
            if (i != 0) throw new ArgumentOutOfRangeException();
            Content = node;
        }
    }
}

File: php/src/Nodes/HtmlNode.php
<?php

namespace LogicAndTrick\WikiCodeParser\Nodes;

class HtmlNode implements INode
{
    public string $htmlBefore;
    public INode $content;
    public string $htmlAfter;

    public string $plainBefore;
    public string $plainAfter;
    public bool $isBlockNode;

    public function __construct(string $htmlBefore, INode $content, string $htmlAfter)
    {
        $this->htmlBefore = $htmlBefore;
        $this->content = $content;
        $this->htmlAfter = $htmlAfter;
        $this->plainBefore = $this->plainAfter = '';
        $this->isBlockNode = false;
    }


    function ToHtml(): string
    {
        return $this->htmlBefore . $this->content->ToHtml() . $this->htmlAfter;
    }

    function ToPlainText(): string
    {
        return $this->plainBefore . $this->content->ToPlainText() . $this->plainAfter;
    }

    function GetChildren(): array
    {
        return  [ $this->content ];
    }

    function ReplaceChild(int $i, INode $node): void
    {
        if ($i !== 0) throw new \InvalidArgumentException("Index out of range");
        $this->content = $node;
    }

    function HasContent(): bool
    {
        return true;
    }
}
